slides/k8s/portworx.md

```diff
@@ -0,0 +1,638 @@
+# Highly available Persistent Volumes
+
+- How can we achieve true durability?
+
+- How can we store data that would survive the loss of a node?
+
+--
+
+- We need to use Persistent Volumes backed by highly available storage systems
+
+- There are many ways to achieve that:
+
+  - leveraging our cloud's storage APIs
+
+  - using NAS/SAN systems or file servers
+
+  - distributed storage systems
+
+--
+
+- We are going to see one distributed storage system in action
+
+---
+
+## Our test scenario
+
+- We will set up a distributed storage system on our cluster
+
+- We will use it to deploy a SQL database (PostgreSQL)
+
+- We will insert some test data in the database
+
+- We will disrupt the node running the database
+
+- We will see how it recovers
+
+---
+
+## Portworx
+
+- Portworx is a *commercial* persistent storage solution for containers
+
+- It works with Kubernetes, but also Mesos, Swarm ...
+
+- It provides [hyper-converged](https://en.wikipedia.org/wiki/Hyper-converged_infrastructure) storage
+
+  (=storage is provided by regular compute nodes)
+
+- We're going to use it here because it can be deployed on any Kubernetes cluster
+
+  (it doesn't require any particular infrastructure)
+
+- We don't endorse or support Portworx in any particular way
+
+  (but we appreciate that it's super easy to install!)
+
+---
+
+## A useful reminder
+
+- We're installing Portworx because we need a storage system
+
+- If you are using AKS, EKS, GKE ... you already have a storage system
+
+  (but you might want another one, e.g. to leverage local storage)
+
+- If you have setup Kubernetes yourself, there are other solutions available too
+
+  - on premises, you can use a good old SAN/NAS
+
+  - on a private cloud like OpenStack, you can use e.g. Cinder
+
+  - everywhere, you can use other systems, e.g. Gluster, StorageOS
+
+---
+
+## Portworx requirements
+
+- Kubernetes cluster ✔️
+
+- Optional key/value store (etcd or Consul) ❌
+
+- At least one available block device ❌
+
+---
+
+## The key-value store
+
+- In the current version of Portworx (1.4) it is recommended to use etcd or Consul
+
+- But Portworx also has beta support for an embedded key/value store
+
+- For simplicity, we are going to use the latter option
+
+  (but we could use the Consul service that we deployed earlier, too)
```

Comment:
This introduces a dependency on a previous module - are we positive both will be used? Maybe say "if deployed"?

Reply:
Good point. I was expecting that both modules would be used together (since the concepts are closely related) but I'll change the wording to avoid having a hard dependency.